### PR TITLE
web export functionality for multiple tasks or all tasks of a course

### LINKED ISSUE #1057: [WEB] Export functionality for multiple tasks or all tasks of a course
ref  #979

### PR DESCRIPTION
**Pull request by nicosoethe**
resolves #1057

### CODE REVIEW

**Review by mxsph — CHANGES_REQUESTED** [review]
The current implementation downloads multiple files when multiple tasks are selected, which is impractical for a user. The backend provides routes to export multiple tasks into a single file, [`/courses/{cid}/tasks/export`](https://thm-mni-ii.github.io/feedbacksystem/api-docs/#tag/Task/operation/exportTasks) and [`/courses/{cid}/tasks/export/all`](https://thm-mni-ii.github.io/feedbacksystem/api-docs/#tag/Task/operation/exportAllTasks), which should be used by the frontend.

**Comment by nicosoethe** [comment]
> The current implementation downloads multiple files when multiple tasks are selected, which is impractical for a user. The backend provides routes to export multiple tasks into a single file, [`/courses/{cid}/tasks/export`](https://thm-mni-ii.github.io/feedbacksystem/api-docs/#tag/Task/operation/exportTasks) and [`/courses/{cid}/tasks/export/all`](https://thm-mni-ii.github.io/feedbacksystem/api-docs/#tag/Task/operation/exportAllTasks), which should be used by the frontend.

Multiple tasks will now be downloaded in one file. I did not see those request options in the api-docs while I was working on this issue 2 weeks ago.

File: modules/fbs-core/web/src/app/dialogs/export-tasks-dialog/export-tasks-dialog.component.ts [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+  export() {
+    this.dialogRef.close({ success: true });
+    let selectedTasks = [];
+    for (let task of this.tasks) {
+      if (task.selected) {
+        selectedTasks.push(task);
+      }
+    }
+    console.log(selectedTasks);
```
Comment (Medienkunzt) [inline modules/fbs-core/web/src/app/dialogs/export-tasks-dialog/export-tasks-dialog.component.ts:30]: remove console.log

File: modules/fbs-core/web/src/app/dialogs/export-tasks-dialog/export-tasks-dialog.component.ts [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+      for (let task of selectedTasks) {
+        taskIds.push(task.id);
+      }
+      this.taskService.downloadMultipleTasks(this.data.courseId, taskIds);
+    }
+    /*     for (let task of this.tasks) {
+      if (task.selected) {
+        this.taskService.downloadTask(this.data.courseId, task.id, task.name);
+      }
+    } */
```
Comment (Medienkunzt) [inline modules/fbs-core/web/src/app/dialogs/export-tasks-dialog/export-tasks-dialog.component.ts:51]: if this part is not necessary, please remove

**Review by Medienkunzt — CHANGES_REQUESTED** [review]

**Review by Medienkunzt — APPROVED** [review]

**Review by mxsph — APPROVED** [review]

**Pull request merged**